packages/sample/src/main.ts

```diff
@@ -568,18 +569,157 @@ program
     .command("servo <channel> <pulseWidth> [frameWidth]")
     .description("Run a servo with pulse width and optional frame width")
     .action(async (channel, pulseWidth, frameWidth) => {
+        let [hub, close] = await getExpansionHubOrThrow();
         let channelValue = Number(channel);
         let pulseWidthValue = Number(pulseWidth);
         let frameWidthValue = frameWidth ? Number(frameWidth) : 4000;
-        let hubs = await openConnectedExpansionHubs();
-        let hub = hubs[0];
-
         runOnSigint(async () => {
             await hub.setServoEnable(channelValue, false);
-            hub.close();
+            close();
         });
 
         await runServo(hub, channelValue, pulseWidthValue, frameWidthValue);
     });
 
 program.parse(process.argv);
+
+/**
+ * Returns the expansion hub referred to by the options provided to the program.
+ * This method also returns a close method. Other hubs may need to be opened, so
+ * prefer calling the returned close method over closing the hub directly.
+ */
+async function getExpansionHubOrThrow(): Promise<[hub: ExpansionHub, close: () => void]> {
+    let options = program.opts();
+    let serialNumber = options.serial;
+    // options.address is a string, so a specified address of "0" will be treated as truthy, and will not be ignored.
+    let moduleAddress = options.address ? Number(options.address) : undefined;
+    let parentAddress = options.parent ? Number(options.parent) : undefined;
+    if (moduleAddress !== undefined && (moduleAddress < 1 || moduleAddress > 255)) {
+        throw new Error(`${moduleAddress} is not a valid module address`);
+    } else if (
+        parentAddress !== undefined &&
+        (parentAddress < 1 || parentAddress > 255)
+    ) {
+        throw new Error(`${parentAddress} is not a valid parent address`);
+    }
+
+    if (moduleAddress !== undefined && parentAddress === undefined) {
+        throw new Error("A module address cannot be specified without a parent address.");
+    }
+
+    if (serialNumber !== undefined) {
+        if (parentAddress === undefined) {
+            throw new Error(
+                "parent address must be specified if serial number is specified.",
+            );
+        }
+        return openExpansionHubWithSerialNumber(
+            serialNumber,
+            parentAddress,
+            moduleAddress,
+        );
+    } else if (parentAddress !== undefined) {
+        return openExpansionHubWithAddress(parentAddress, moduleAddress);
+    }
+
+    let connectedHubs: ParentExpansionHub[] = await openConnectedExpansionHubs();
+    if (connectedHubs.length == 0) {
+        throw new Error("No hubs are connected");
+    }
+    if (connectedHubs.length > 1 || connectedHubs[0].children.length > 0) {
+        throw new Error(
+            "Multiple hubs connected. You must specify a serialNumber and/or address.",
+        );
+    }
+
+    // The user did not specify a specific Hub to open. Open all of them.
```

Comment:
I previously didn't notice the check for multiple hubs that comes right before this point, sorry
```suggestion
    // Open the only Hub that is connected
```